react-app/src/components/users/UserDetail.tsx

```diff
@@ -27,8 +28,9 @@ interface UserProfile extends User {
 
 const UserDetail = ({ onClose }: IUserDetail) => {
   const { id } = useParams();
-  const { pimsUser } = useContext(AuthContext);
+  const user = useContext(AuthContext);
   const api = usePimsApi();
+  const pimsUser = { user };
```

Comment:
This change is creating an extra layer to this object, which is why you needed the change on line 139. 
useContext will give back something like { pimsUser, keycloakUser }, so the original used destructuring to get pimsUser directly.
Would recommend keeping the original, but just getting keycloakUser through destructuring as well so you can use it for the hasRoles function below.

Reply:
added changes in newest commit